pkg/provision/workspace/deployment.go

```diff
@@ -169,6 +169,35 @@ func ScaleDeploymentToZero(ctx context.Context, workspace *common.DevWorkspaceWi
 	return nil
 }
 
+// Returns an error message and a FailureReason if the workspace related pods are in an unrecoverable state, which may
+// have been caused  by an ignoredUnrecoverableEvent.
+// Otherwise, empty strings are returned for the error message and failure reason.
+func CheckForIgnoredWorkspacePodEvents(workspace *common.DevWorkspaceWithConfig, clusterAPI sync.ClusterAPI) (errMessage string, reason metrics.FailureReason) {
```

Comment:
Why should this function return a `metrics.FailureReason`? The devworkspace controller already has the machinery to figure out the failure reason based on a string describing the failure.